File: School Classes/Programming 2(Java)/Module_4/src/lab1/Car.java
//************************************************************************
// Author: Lane Pollock
// Date: 22 Oct 2024
// Language: Java
// Assignment: Lab 4.1
// Description: Car Class to hold make, model, year, price of a car
//************************************************************************


package lab1;


public class Car {
	//instance variables
	private String make, model;
	private int year;
	private double price;
	
	//constructors
	
	public Car() {
		make = "unknown";
		model = "unknown";
		year = 0;
		price = 0.0;
	}
	
	public Car(String make, String model, int year, double price) {
		this.make = make;
		this.model = model;
		this.year = year;
		this.price = price;
	}
	
	//getter/setters

	public void setMake(String make) {
		this.make = make;
	}
	public String getMake() {
		return make;
	}
	public String getModel() {
		return model;
	}
	public void setModel(String model) {
		this.model = model;
	}
	public int getYear() {
		return year;
	}
	public void setYear(int year) {
		this.year = year;
	}
	public double getPrice() {
		return price;
	}
	public void setPrice(double price) {
		this.price = price;
	}

	
}

File: School Classes/Programming 1(dev c++)/CarProject/Car.cpp
//************************************************************************
//	Author: Lane Pollock	
//	Date: 24 Apr 2024
//	Language: C++
//	Assignment: 
//	Description: implementation file for car class
//		car.cpp
//************************************************************************

#include "Car.hpp"
#include <string>
#include <iomanip>

using namespace std;

//define the getter methods
string Car::getMake() const
{
	return make;
}

string Car::getModel() const
{
	return model;
}

int Car::getYear() const
{
	return year;
}

double Car::getPrice() const
{
	return price;
}

//define the setter methods
void Car::setMake(string pMake)
{
	make = pMake;
}

void Car::setModel(string pModel)
{
	model = pModel;
}

void Car::setYear(int pYear)
{
	year = pYear;
}

void Car::setPrice(double pPrice)
{
	price = pPrice;
}

//class methods
void Car::print()
{
	cout << fixed << showpoint << setprecision(2); //money
	//display car details
	cout << "Make: " << make << endl
		 << "Model: " << model << endl
		 << "Year: " << year << endl
		 << "Price: $" << price << endl;
}

//constructors
//constructors are only called when an instance of a class is made****
Car::Car(string pMake, string pModel, int pYear, double pPrice)
{
	make = pMake;
	model = pModel;
	year = pYear;
	price = pPrice;
}

Car::Car(string pMake, string pModel, int pYear)
{
	make = pMake;
	model = pModel;
	year = pYear;
	price = 0.0;
}

Car::Car()
{
	make = "Unknown Make";
	model = "Unknown Model";
	year = 2023;
	price = 0.0;
}